File: web/src/api/models/Variants/VariantDto.ts
import { ClothesUnitOfMeasure } from '../Products/ClothesUnitOfMeasure';

export interface VariantDto {
  variantId: number;
  productId: number;
  productName: string;
  unitOfSize: ClothesUnitOfMeasure;
  size: number;
}

File: api/DropShot.Application/Products/Models/VariantDto.cs
using DropShot.Domain.Enums;

namespace DropShot.Application.Products.Models;

public class VariantDto
{
    public int VariantId { get; set; }
    public int ProductId { get; set; }
    public string ProductName { get; set; }
    public ClothesUnitOfMeasure UnitOfSize { get; set; }
    public int Size { get; set; }
}